src/main/java/ru/yandex/market/graphouse/retention/MetricRetention.java

```diff
@@ -16,22 +16,71 @@
  * @date 23.11.16
  */
 public class MetricRetention {
-    private final Pattern pattern;
+    private final Pattern mainPattern;
+    private final Pattern secondPattern;
     private final String function;
+    private final boolean isDefault;
     private final RangeMap<Integer, Integer> ranges;
+    public final static int typeRetention = 1;
+    public final static int typeAggregation = 2;
+    public final static int typeAll = 3;
+
+    private MetricRetention(Pattern mainPattern, String function, boolean isDefault) {
+        if (isDefault) {
+            this.mainPattern = null;
+        } else {
+            this.mainPattern = mainPattern;
+        }
+        this.secondPattern = null;
+        this.function = function;
+        this.isDefault = isDefault;
+        this.ranges = TreeRangeMap.create();
+    }
 
-    private MetricRetention(Pattern pattern, String function) {
-        this.pattern = pattern;
+    private MetricRetention(Pattern mainPattern, Pattern secondPattern, String function) {
+        this.mainPattern = mainPattern;
+        this.secondPattern = secondPattern;
         this.function = function;
+        this.isDefault = false;
         this.ranges = TreeRangeMap.create();
     }
 
+    @Override
+    public String toString() {
+        return "Main pattern: " + mainPattern + "; Second pattern: " + secondPattern + "; Function: " + function +
+            "; Ranges: " + ranges.toString();
+    }
+
     public String getFunction() {
         return function;
     }
 
+    public String getMainPattern() { return mainPattern.toString(); }
+
+    public RangeMap<Integer, Integer> getRanges() { return ranges; }
+
+    public boolean getIsDefault() { return isDefault; }
+
+    public int getType() {
+        if (function.equals("")) {
+            return typeRetention;
+        }
+        if (ranges.toString().equals("[[0..+∞)=0]")) {
```

Comment:
I would appreciate a hint on how to check it in the correct way.

Reply:
What are you trying to do here?

Reply:
Checking if Retention has ranges. Done with `null` range during a build, thank you